application/common/models/UserBase.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+namespace common\models;
+
+use Yii;
+
+/**
+ * This is the model class for table "user".
+ *
+ * @property integer $id
+ * @property string $employee_id
+ * @property string $first_name
+ * @property string $last_name
+ * @property string $display_name
+ * @property string $username
+ * @property string $email
+ * @property string $password_hash
+ * @property string $active
+ * @property string $locked
+ * @property string $last_changed_utc
+ * @property string $last_synced_utc
+ *
+ * @property PasswordHistory[] $passwordHistories
+ */
+class UserBase extends \yii\db\ActiveRecord
+{
+    /**
+     * @inheritdoc
+     */
+    public static function tableName()
+    {
+        return 'user';
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function rules()
+    {
+        return [
+            [['employee_id', 'first_name', 'last_name', 'username', 'email', 'active', 'locked', 'last_changed_utc', 'last_synced_utc'], 'required'],
+            [['active', 'locked'], 'string'],
+            [['last_changed_utc', 'last_synced_utc'], 'safe'],
+            [['employee_id', 'first_name', 'last_name', 'display_name', 'username', 'email', 'password_hash'], 'string', 'max' => 255],
+            [['employee_id'], 'unique'],
+            [['username'], 'unique'],
+            [['email'], 'unique'],
+        ];
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function attributeLabels()
+    {
+        return [
+            'id' => Yii::t('app', 'ID'),
+            'employee_id' => Yii::t('app', 'Employee ID'),
+            'first_name' => Yii::t('app', 'First Name'),
+            'last_name' => Yii::t('app', 'Last Name'),
+            'display_name' => Yii::t('app', 'Display Name'),
+            'username' => Yii::t('app', 'Username'),
+            'email' => Yii::t('app', 'Email'),
+            'password_hash' => Yii::t('app', 'Password Hash'),
+            'active' => Yii::t('app', 'Active'),
+            'locked' => Yii::t('app', 'Locked'),
+            'last_changed_utc' => Yii::t('app', 'Last Changed Utc'),
+            'last_synced_utc' => Yii::t('app', 'Last Synced Utc'),
```

Comment:
I like that!  https://github.com/silinternational/idp-id-broker/commit/2c1302ce364ffd3fd7780dac4e2d3316dc3d3de2